components/site/src/lib.rs

```diff
@@ -583,11 +584,29 @@ impl Site {
         imageproc.do_process()
     }
 
-    /// Deletes the `public` directory if it exists
+    /// Deletes the contents in the `public` directory if it exists
     pub fn clean(&self) -> Result<()> {
         if self.output_path.exists() {
-            // Delete current `public` directory so we can start fresh
-            remove_dir_all(&self.output_path).context("Couldn't delete output directory")?;
+            // Delete contents of current `public` directory so we can start fresh
+            for entry in self.output_path.read_dir().context(format!(
```

Comment:
No need to iterate on it if `self.config.preserve_hidden_files` is `false`. You can return the original line from the diff on the left in that case.